### PR TITLE
feat: add gitHooks

### PR DESCRIPTION
**Pull request by Muhesh7**
In this PR

* Github hooks as been added for
   - pre-commit:
       which  runs ```flutter format .``` (Formats the dart code)
   - pre-push:
      checks for uncommitted files and if found any wont let u push to the repo
      followed by ```flutter analyse``` (Checks for Linting errors and we exit the process if any error is detected)
    
* UI modifications are done:
   changed colors and designs of widgets
   
* Increased memory efficiency by disposing controllers  when its corresponding view losses its scope
  done by replacing Get.toNamed() to Get.offAllNamed()

* Code Modularity:
   modularised widgets by creating a seperate file

### CODE REVIEW

**Comment by Sudhindra3000** [comment]
Maybe give separate PRs for githooks and UI changes?

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+echo "Run Formatting..."
+flutter format .
+echo "Run Linting..."
+flutter analyze
```
Comment (Sudhindra3000) [inline .githooks/pre-commit:5]:
```suggestion
echo "Running Formatter..."
flutter format .
echo "Running Analyzer..."
flutter analyze
```

File: .githooks/pre-push [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+if [[ `git status --porcelain` ]]
+then
+  echo "un-commited changes found, do commit or stash them"
+  exit 1
+fi
```
Comment (Sudhindra3000) [inline .githooks/pre-push:6]: Not going to do anything else in pre-push?
Maybe we can shift `flutter analyze` step to here

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: .githooks/pre-push (hunk 2 above)
Comment (Muhesh7) [reply .githooks/pre-push:6]: thought of running running ```flutter test``` in pre-push
but yeah will shift ```flutter analyse``` to pre-push from pre-commit

File: lib/widgets/app_theme.dart [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 class AppTheme {
   static const colours = AppColours();
 
   const AppTheme._();
 
-  static ThemeData define() {
-    return ThemeData();
-  }
+  static ThemeData define() => ThemeData();
 }
```
Comment (Sudhindra3000) [inline lib/widgets/app_theme.dart:11]: Can we can move this file to a separate theme folder as its not a widget

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: lib/widgets/app_theme.dart (hunk 3 above)
Comment (Muhesh7) [reply lib/widgets/app_theme.dart:11]: Ok will do it and create themes folder inside views and migrate widgets folder inside views

**Review by Sudhindra3000 — APPROVED** [review]

**Pull request merged**